smac/facade/smac_facade.py

```diff
@@ -337,6 +338,11 @@ def optimize(self):
             self.logger.info("Final Incumbent: %s" % (self.solver.incumbent))
             self.runhistory = self.solver.runhistory
             self.trajectory = self.solver.intensifier.traj_logger.trajectory
+
+            if self.solver.scenario.output_dir is not None:
```

Comment:
why not in a `finally ` as I did in the cli?
If we raised a  TAEAbortException (or something else crashed), we maybe still want to have the data.